### ISSUE TITLE
Restart `neofs-ir` aplication on errors

### ISSUE THREAD

**Issue by fyrchik** [nspcc-dev/neofs-node#1140]
https://github.com/nspcc-dev/neofs-node/blob/master/cmd/neofs-ir/main.go#L96
Maybe this should be done via external means and not in code.

**Comment by alexvanin** [comment]
> Maybe this should be done via external means and not in code.

Agree. This was related more to `listener` component errors which produced on Neo RPC node restart. As it proposed in https://github.com/nspcc-dev/neofs-node/issues/1053#issuecomment-1015316718, with single client instance, we can switch between endpoints in runtime. 

However in https://github.com/nspcc-dev/neofs-node/issues/157#issue-737602743 I specified some issues during endpoint switch in runtime. It requires some sort of state resync after endpoint transition. Maybe it is not worth to implement resync routine in code and just restart the whole app by external entity.

Anyway, blocked by #1053 for now.

**Comment by alexvanin** [comment]
Let's drop this `todo` in https://github.com/nspcc-dev/neofs-node/pull/1145

**Comment by fyrchik** [comment]
Closed via #1145 .
